File: Data Structures & Algorithms/maximum-subarray/submission-0.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        """
            kadane's algo
        """
        n = len(nums)
        
        dp = [0] * n
        result = nums[0]
        arr_sum = 0

        for i in range(0,n):
            dp[i] = max(nums[i], nums[i] + dp[i-1])
            result = max(result,dp[i])

        return result

File: Data Structures & Algorithms/longest-increasing-subsequence/submission-0.cpp
class Solution {
public:
    int lengthOfLIS(vector<int>& nums) {
        /*
            9 1 4 2 3 3 7
            1 1 2 2 1 1 1
        */
        int n = nums.size();
        vector<int>dp(n,1);
        int result = 1;

        //O(n^2)
        for(int i=0; i<n; i++){
            for(int j=i-1; j>=0; j--){
                if(nums[i] > nums[j]){
                    dp[i] = max(dp[i],1 + dp[j]);
                }
            }
            result = max(result,dp[i]);
        }

        return result;

    }
};
